app/src/androidTest/java/com/github/versus/OptionChooseLocationTest.java

```diff
@@ -51,69 +52,51 @@ public class OptionChooseLocationTest {
         onView(withId(R.id.drawer_layout)).check(matches(DrawerMatchers.isOpen(GravityCompat.START)));
         onView(withId(R.id.nav_location)).perform(click());
     }
 
+    /**
+     * Cleans up the testing environment after each test.
+     * Releases the Intents framework.
+     */
+    @After
+    public void tearDown() {
+        Intents.release();
+    }
+
+
     /**
      * Test for simulating a click on the map.
      * Ensures that clicking on the map results in the expected behavior.
      *
      * @throws InterruptedException if the test is interrupted
      */
-    //@Test
+    @Test
     public void testClick() throws InterruptedException {
+        long waitingTime = 6000;
+        ElapsedTimeIdlingResource idlingResourceFirst = new ElapsedTimeIdlingResource(waitingTime);
+
         String placeName = "GooglePlex Football";
         Espresso.openActionBarOverflowOrOptionsMenu(ApplicationProvider.getApplicationContext());
 
         // Open the Choose location option
         onView(withText("Choose location")).perform(click());
-        
-        // Get screen dimensions
-        DisplayMetrics displayMetrics = new DisplayMetrics();
-        WindowManager windowManager = (WindowManager) ApplicationProvider.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
-        windowManager.getDefaultDisplay().getRealMetrics(displayMetrics);
-
-        // Calculate the center coordinates
-        int centerX = displayMetrics.widthPixels / 2;
-        int centerY = displayMetrics.heightPixels / 2;
-
         // Simulate a map click event at the center of the screen
-        simulateMapClick(centerX, centerY);
-        onView(withText(placeName)).perform(click());
-    }
+        simulateMapClick(540, 1170);
 
-    //@Test
-    public void testDrawingPathWithPlaceSelected() throws InterruptedException {
-        String placeName = "GooglePlex Football";
-        Espresso.openActionBarOverflowOrOptionsMenu(ApplicationProvider.getApplicationContext());
-        
-        // Open the Choose location option
-        onView(withText("Choose location")).perform(click());
-        
-        // Get screen dimensions
-        DisplayMetrics displayMetrics = new DisplayMetrics();
-        WindowManager windowManager = (WindowManager) ApplicationProvider.getApplicationContext().getSystemService(Context.WINDOW_SERVICE);
-        windowManager.getDefaultDisplay().getRealMetrics(displayMetrics);
-        
-        // Calculate the center coordinates
-        int centerX = displayMetrics.widthPixels / 2;
-        int centerY = displayMetrics.heightPixels / 2;
-        
-        // Simulate a map click event at the center of the screen
-        simulateMapClick(centerX, centerY);
-        
+
+        long waitingTime2 =5000;
+        ElapsedTimeIdlingResource idlingResource2 = new ElapsedTimeIdlingResource(waitingTime2);
+        IdlingRegistry.getInstance().register(idlingResource2);
```

Comment:
I'm not sure that these tests uses correctly IdlingResources (this might break the CI again)

Reply:
I think it works fine , but we can discuss this more in detail with the team !